docs/docs/reference/built-in-components/gatsby-script.md

```diff
@@ -0,0 +1,282 @@
+- Via a template literal
+
+Here's a look at both:
+
+```jsx
+<Script id="first-unique-id" dangerouslySetInnerHTML={{ __html: `alert('Hello world')` }} />
+<Script id="second-unique-id">{`alert('Hello world')`}</Script>
+```
+
+Functionally, both of these ways of defining inline scripts are equivalent.
+
+## Strategies
+
+You can declare a loading strategy by passing a `strategy` property. These are the available loading strategies:
+
+- `post-hydrate` - Loads after the page has hydrated
+- `idle` - Loads after the page has become idle
+- `off-main-thread` - Loads off the main thread in a web worker via [Partytown](https://partytown.builder.io)
+
+The best strategy to use depends on the functionality of the script you would like to include.
+
+A general rule of thumb is if your script can afford to be loaded later (as many analytics scripts can be), then start with `off-main-thread` or `idle` and move to earlier loading strategies if necessary.
+
+Here's how you can define these strategies in the `<Script>` component:
+
+```jsx
+<Script src="https://my-example-script" strategy="post-hydrate" />
+<Script src="https://my-example-script" strategy="idle" />
+<Script src="https://my-example-script" strategy="off-main-thread" />
+```
+
+Additionally, Gatsby exports a `ScriptStrategy` enum that you can use in TSX files if you prefer:
+
+```tsx
+import { Script, ScriptStrategy } from "gatsby"
+
+<Script src="https://my-example-script" strategy={ScriptStrategy.postHydrate} />
+<Script src="https://my-example-script" strategy={ScriptStrategy.idle} />
+<Script src="https://my-example-script" strategy={ScriptStrategy.offMainThread} />
+```
+
+### Off main thread strategy
+
+The `off-main-thread` strategy leverages [Partytown](https://partytown.builder.io) under the hood, so it requires further configuration. See the [Partytown configuration](https://partytown.builder.io/configuration) documentation for complete details.
+
+Here is an example configuring the `<Script>` component with [Google Analytics](https://analytics.google.com/analytics/web/):
+
+```tsx
+import { Script, ScriptStrategy } from "gatsby"
+
+// `process.env.GTM` is your Google Analytics 4 identifier defined in your `.env.production` and `.env.development` files
+
+<Script
+  src={`https://www.googletagmanager.com/gtag/js?id=${process.env.GTM}`}
+  strategy={ScriptStrategy.offMainThread}
+  forward={[`gtag`]}
+/>
+<Script id="gtag-config" strategy={ScriptStrategy.offMainThread}>
+  {`
+    window.dataLayer = window.dataLayer || [];
+    window.gtag = function gtag() { window.dataLayer.push(arguments); }
+    gtag('js', new Date());
+    gtag('config', ${process.env.GTM}, { send_page_view: false })
+  `}
+</Script>
+```
+
+#### Forward collection
+
+Gatsby will collect all `off-main-thread` scripts on a page, and automatically merge any [Partytown forwarded events](https://partytown.builder.io/forwarding-events) defined via the `forward` property into a single configuration for each page:
```

Comment:
should we be more explicit here and mention that this is only working on Gatsby Cloud?

Reply:
Forward collection happens during SSR in the build process, so this works on any hosting service since it's in the output HTML